Tests/Unit/LoRaWan/LoRaDataRateTests.cs

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace LoRaWan.Tests.Unit
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using LoRaWan;
+    using LoRaWan.Tests.Common;
+    using Xunit;
+
+    public class LoRaDataRateTests
+    {
+        private static IEnumerable<T> LoRaDataRates<T>(Func<SpreadingFactor, Bandwidth, T> selector) =>
+            from sf in Enum.GetValues<SpreadingFactor>()
+            from bw in Enum.GetValues<Bandwidth>()
+            select selector(sf, bw);
+
+        public static readonly TheoryData<SpreadingFactor, Bandwidth> FromData =
+            TheoryDataFactory.From(LoRaDataRates(ValueTuple.Create));
+
+        [Theory]
+        [MemberData(nameof(FromData))]
+        public void From(SpreadingFactor sf, Bandwidth bw)
+        {
+            var subject = LoRaDataRate.From(sf, bw);
+
+            Assert.Equal(sf, subject.SpreadingFactor);
+            Assert.Equal(bw, subject.Bandwidth);
+            Assert.Equal(ModulationKind.LoRa, subject.ModulationKind);
+        }
+
+        public static readonly TheoryData<string, SpreadingFactor, Bandwidth> ParseFormatData =
+            TheoryDataFactory.From(LoRaDataRates((sf, bw) => ($"{sf}{bw}", sf, bw)));
+
+        [Theory]
+        [MemberData(nameof(ParseFormatData))]
+#pragma warning disable xUnit1024 // Test methods cannot have overloads
+        public void ToString(string expected, SpreadingFactor sf, Bandwidth bw)
```

Comment:
nit: instead of disabling the warning, what about colling the test method "ToString_Returns_Expected_Format" or something similar?